draft-add-requirements.md

```diff
@@ -142,18 +271,97 @@ Narrative required.
+
+Only sending the necessary queries through the enterprise resolver, and not generic Internet queries, has the privacy
+benefit of only exposing traffic to the enterprise that fall within a limited set of domains.
+
+Using encrypted DNS for private names also opens up the possibility of doing private name resolution outside
+of the content of a VPN or managed network. If the DNS resolver authenticates clients, it can offer its resolver
+for private names on a publicly accessible server, while still limiting the visibility of the DNS traffic.
 
 | Requirement | Description |
 | R11.1 | Example requirement |
 
-## Discover a mapping between a public domain and a resolver {#last-usecase}
+## Encrypted resolvers for content providers {#cdn-content}
 
-Narrative required.
+Content Delivery Networks (CDNs), and content-providers more broadly, can also provide encrypted DNS resolvers that can
+be used by clients over the public Internet. These resolvers can either allow resolution of all public names (like
+normal recursive resolvers), or be designed to serve a subset of names managed by the content provider (like an
+authoritative resolver). Using these resolvers can allow the content provider to directly control how DNS answers are
+used for load balancing and address selection, which could improve performance of connections to the content provider.
+
+Using a content-provider's encrypted resolver can also provide several privacy and security benefits: 
+
+- Prevent devices along the network path to the content-provider's resolver from observing client DNS messages
+- Verify that answers come from the entity that manages the domains being resolved
+- Reduce the number of entities able to monitor the specific names accessed by a client to only the client and the content provider, assuming that the content provider would already see the names upon a secure connection later being made based on the DNS answers (e.g., in the TLS SNI extension)
 
 | Requirement | Description |
 | R12.1 | Example requirement |
 
+# Privacy and security requirements {#priv-sec}
+
+Encrypted DNS improves the privacy and security of DNS queries and answers in the presence of malicious
+attackers. Such attackers are assumed to interfere with or otherwise impede DNS traffic and corresponding
+discovery mechanisms. They may be on-path or off-path between the client and entities with which the client
+communicates {{?RFC3552}}. These attackers can inject, tamper, or otherwise interfere with traffic as needed.
+Given these capabilities, an attacker may have a variety of goals, including, though not limited to:
+
+- Monitor and profile clients by observing unencrypted DNS traffic
+
+- Modify unencrypted DNS traffic to filter or augment the user experience
+
+- Block encrypted DNS
+
+Clients cannot assume that their network does not have such an attacker unless given some means of authenticating or otherwise
+trusting the communication with their DNS resolver.
```

Comment:
Trusting the communication is only possible by authenticating the DNS resolver.

Reply:
I guess "blind trust" is another option; people frequently assume that there is no attacker when doing Do53 or opportunistic DoT. This is not particularly wise but it happens, particularly in networks where there is no history of attack.

Reply:
Enterprise networks with best security practices are still susceptible to internal attacks (see https://tools.ietf.org/html/draft-arkko-farrell-arch-model-t-03). Enterprises have started migrating to "zero trust security" framework to encrypt all communication to minimize the threat (https://www.cloudflare.com/learning/security/glossary/what-is-zero-trust/. If internal attacks are not an issue, Enterprises would not move to using HTTPS for internal servers.

I propose the following change:
Clients cannot assume that their network does not have such an attacker, encryption and authentication of DNS resolver is required to defend from active and passive attacks.

Reply:
Indeed, Zero Trust is being adopted in enterprises. Let's move this to an issue so we can review the wording properly and more widely.